src/ngx_http_lua_ssl.c

```diff
@@ -34,4 +33,112 @@ ngx_http_lua_ssl_init(ngx_log_t *log)
 }
 
 
+int
+ngx_http_lua_ssl_password_callback(char *buf, int size, int rwflag,
+    void *userdata)
+{
+    ngx_str_t *pwd = userdata;
+
+    if (rwflag) {
+        ngx_log_error(NGX_LOG_ALERT, ngx_cycle->log, 0,
+                      "ngx_http_lua_ssl_password_callback() "
+                      "is called for encryption");
+        return 0;
+    }
+
+    if (pwd->len == 0) {
+        return 0;
+    }
+
+    if (pwd->len > (size_t) size) {
+        ngx_log_error(NGX_LOG_ERR, ngx_cycle->log, 0,
+                      "password is truncated to %d bytes", size);
+
+    } else {
+        size = pwd->len;
+    }
+
+    ngx_memcpy(buf, pwd->data, size);
+
+    return size;
+}
+
+
+ngx_int_t
+ngx_http_lua_ssl_certificate(ngx_ssl_t *ssl, ngx_str_t *cert,
+    ngx_str_t *priv_key, ngx_str_t *password, ngx_log_t *log)
+{
+    BIO                     *cbio = NULL;
+    BIO                     *pbio = NULL;
+    X509                    *x509 = NULL;
+    EVP_PKEY                *pkey = NULL;
+    ngx_int_t                rc = NGX_ERROR;
+
+    cbio = BIO_new_mem_buf((char *)cert->data, cert->len);
+    if (cbio == NULL) {
+        ngx_ssl_error(NGX_LOG_ERR, log, 0, "BIO_new_mem_buf() failed");
+        goto done;
+    }
+
+    /*
+    * Reading the PEM-formatted certificate from memory into an X509
+    */
+
+    x509 = PEM_read_bio_X509(cbio, NULL, 0, NULL);
```

Comment:
Given that we already have PEM (and DER) parsing functions, should users maybe use those and pass the already parsed chain to this function?